remove references to --manifests command in generate packagemanifest docs

komish opened the pull request:
Signed-off-by: Jose R. Gonzalez <josegonzalez89@gmail.com>

**Description of the change:**
Quick change to docs which makes reference to a flag `--manifests` which doesn't seem to exist in the `generate packagemanifests` action anymorew.

```
Flags:
      --channel string         Channel name for the generated package
      --crds-dir string        Root directory for CustomResoureDefinition manifests
      --default-channel        Use the channel passed to --channel as the package manifest file's default channel
      --deploy-dir string      Root directory for operator manifests such as Deployments and RBAC, ex. 'deploy'. This directory is different from that passed to --input-dir
      --from-version string    Semantic version of the operator being upgraded from
  -h, --help                   help for packagemanifests
      --input-dir string       Directory to read existing package manifests from. This directory is the parent of individual versioned package directories, and different from --deploy-dir
      --kustomize-dir string   Directory containing kustomize bases and a kustomization.yaml for operator-framework manifests (default "config/manifests")
      --output-dir string      Directory in which to write package manifests
  -q, --quiet                  Run in quiet mode
      --stdout                 Write package to stdout
      --update-objects         Update non-CSV objects in this package, ex. CustomResoureDefinitions, Roles (default true)
  -v, --version string         Semantic version of the packaged operator
```

**Motivation for the change:**
Cleaner docs!


**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [ ] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
- [X] Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)

camilamacedo86 reviewed (changes requested):
Really tks for your contribution 🥇. 

However, see that the docs are generated automatically and checked in the CI.
The makefile command used to do that is [gen-cli-doc](
https://github.com/operator-framework/operator-sdk/blob/master/Makefile#L109). 


So, if you run this command will you check its result or not?  Could you please use the command and let us know?

komish commented:
@camilamacedo86 Thanks for the guidance! 

I ran the make target `gen-cli-doc` and didn't see anything change in the repository. Am I making the change in the wrong place? 

I did notice Travis reported a failure due to a broken link in another file (unrelated to this PR). I've fixed that, though I've made an assumption as to what file should have been linked. Let me know if I've assumed incorrectly and I can make the change.

camilamacedo86 commented:
Hi @komish, 

The `make gen-cli-doc` is responsable to update the docs. The same actions are performed to ensure that the docs are updated. The same script to gen is called in https://github.com/operator-framework/operator-sdk/blob/96d8bae49941cfe82076c9357366425f7e51d6be/hack/tests/sanity-check.sh#L10 and we check if it has diff. 

So, if the docs are outdated and it's passing with the changes made here and not then, we might have a problem in the script used to gen it and/or in the CI. We would need to check it better.

komish commented:
@camilamacedo86 couple of things:

I reverted the change I made to the `website/content/en/docs/contribution-guidelines/testing/travis-build.md` - originally the tests failed because this file had a link that was throwing a 404. That doesn't seem to be the case now, so I've reverted that change and squashed the git history. Sorry for the confusion there!

> So, if the docs are outdated and it's passing with the changes made here and not then, we might have a problem in the script used to gen it and/or in the CI. We would need to check it better.

I think the script is generating as I expect, but I've included the changes the make target would make to the website files in the commit. Should I remove those changes from the commit? 

I tested by reverting the website file `website/content/en/docs/cli/operator-sdk_generate_packagemanifests.md` back to what it was prior to the commit and ran the `make` target. It rendered a clean working tree.

```
 $ git status
On branch fix-docs-generate-packagemanifests
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   website/content/en/docs/cli/operator-sdk_generate_packagemanifests.md

no changes added to commit (use "git add" and/or "git commit -a")
```

```
 $ make gen-cli-doc 
./hack/generate/cli-doc/gen-cli-doc.sh
```

```
 $ git status
On branch fix-docs-generate-packagemanifests
nothing to commit, working tree clean
```

---
Either way, let me know what I can help with further!

estroz reviewed (approved):
/lgtm
